AutomatedTesting/Gem/PythonTests/scripting/Node_HappyPath_DuplicateNode.py

```diff
@@ -34,80 +46,67 @@ def Node_HappyPath_DuplicateNode():
-    helper.wait_for_condition(lambda: general.is_pane_visible("Script Canvas"), 5.0)
-
-    # 2) Open a new graph
-    editor_window = pyside_utils.get_editor_main_window()
-    sc = editor_window.findChild(QtWidgets.QDockWidget, "Script Canvas")
-    sc_main = sc.findChild(QtWidgets.QMainWindow)
-    create_new_graph = pyside_utils.find_child_by_pattern(
-        sc_main, {"objectName": "action_New_Script", "type": QtWidgets.QAction}
-    )
-    if sc.findChild(QtWidgets.QDockWidget, "NodeInspector") is None:
-        action = pyside_utils.find_child_by_pattern(sc, {"text": "Node Inspector", "type": QtWidgets.QAction})
-        action.trigger()
-    node_inspector = sc.findChild(QtWidgets.QDockWidget, "NodeInspector")
-    create_new_graph.trigger()
-
-    # 3) Add node
-    command_line_input("add_node Print")
-
-    # 4) Duplicate node
-    graph_view = sc.findChild(QtWidgets.QFrame, "graphicsViewFrame")
-    graph = graph_view.findChild(QtWidgets.QWidget, "")
-    # There are currently no utilities available to directly duplicate the node,
-    # therefore the node is selected using CTRL+A on the graph to select
-    # it and then CTRL+D to duplicate
-    sc_main.activateWindow()
-    QtTest.QTest.keyClick(graph, "a", Qt.ControlModifier, WAIT_FRAMES)
-    QtTest.QTest.keyClick(graph, "d", Qt.ControlModifier, WAIT_FRAMES)
-
-    # 5) Verify the node was duplicated
-    # As direct interaction with node is not available the text on the label
-    # inside the Node Inspector is validated showing two nodes exist
-    after_dup = grab_title_text()
-    Report.result(Tests.node_duplicated, after_dup == EXPECTED_STRING)
-
-
-if __name__ == "__main__":
-    import ImportPathHelper as imports
-
-    imports.init()
-    from editor_python_test_tools.utils import Report
-
-    Report.start_test(Node_HappyPath_DuplicateNode)
+    @pyside_utils.wrap_async
+    async def run_test(self):
+
+        # Preconditions
+        general.idle_enable(True)
+
+        # 1) Open Script Canvas window (Tools > Script Canvas)
+        general.open_pane(SCRIPT_CANVAS_UI)
+        helper.wait_for_condition(lambda: general.is_pane_visible(SCRIPT_CANVAS_UI), WAIT_TIME_3)
+
+        # 2) Open a new graph
+        editor_window = pyside_utils.get_editor_main_window()
+        sc_editor = editor_window.findChild(QtWidgets.QDockWidget, SCRIPT_CANVAS_UI)
+        sc_editor_main_window = sc_editor.findChild(QtWidgets.QMainWindow)
+        create_new_graph = pyside_utils.find_child_by_pattern(
+            sc_editor_main_window, {"objectName": "action_New_Script", "type": QtWidgets.QAction}
+        )
+        if sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT) is None:
+            action = pyside_utils.find_child_by_pattern(sc_editor, {"text": NODE_INSPECTOR_UI, "type": QtWidgets.QAction})
+            action.trigger()
+        sc_graph_node_inspector = sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT)
+        create_new_graph.trigger()
```

Comment:
Let's extract another util definition here.

```
def create_new_sc_graph(...)
       pyside_utils.find_child_by_pattern(
            sc_editor_main_window, {"objectName": "action_New_Script", "type": QtWidgets.QAction}
        )
        if sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT) is None:
            action = pyside_utils.find_child_by_pattern(sc_editor, {"text": NODE_INSPECTOR_UI, "type": QtWidgets.QAction})
            action.trigger()
        sc_graph_node_inspector = sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT)
        create_new_graph.trigger()
```

We can then write the following code to simplify the test script as

`new_sc_graph = create_new_sc_graph(...)`

Reply:
broke this chunk of the test into 2 functions since there are two separate actions happening here; opening a new file and getting the inspector widget.